### ISSUE TITLE
Feature request: Improve foot routing by supporting additional sidewalk values

### ISSUE THREAD

**Issue by scaidermern** [osmandapp/Osmand#2567]
Similar to cycleway:left and cycleway:right (issue #2566) there are additional values to the [sidewalk key](https://wiki.openstreetmap.org/wiki/Sidewalks). These are for example _sidewalk=none_ ([used 255k times](https://taginfo.openstreetmap.org/tags/sidewalk=none)) and the synonym _sidewalk=no_ ([used 120k times](https://taginfo.openstreetmap.org/tags/sidewalk=no)), _sidewalk=both_ ([used 227k times](https://taginfo.openstreetmap.org/tags/sidewalk=both)), _sidewalk=right_ ([used 111k times](https://taginfo.openstreetmap.org/tags/sidewalk=right)), _sidewalk=left_ ([used 49k times](https://taginfo.openstreetmap.org/tags/sidewalk=left)) and _sidewalk=separate_ ([used 12k times](https://taginfo.openstreetmap.org/tags/sidewalk=separate)).

Proper handling of these values could look like this:

- Assign a really low priority to ways tagged as _sidewalk=none_ and _sidewalk=no_ (a synonym) because there is no sidewalk available and walking on the road is usually dangerous. Ideally this low priority may also differ depending on the type of the road since walking on a primary road without sidewalk is way more dangerous than walking on a residential road. But I think the latter suggestion is like putting the cherry on the cake and not really needed.
- Assign the same low priority to ways tagged with _sidewalk=separate_. This tag means there *is* a sidewalk available, but it is mapped as a separate way. Thus the current way should *not* be used for routing.
- Assign a high priority to ways tagged as _sidewalk=both_. This priority should be higher than for ways without a sidewalk tag in order to prefer safe routes.
- Assign the same high priority to ways tagged with _sidewalk=left_ or _sidewalk=right_, but only to the corresponding direction. Not sure how difficult this is to implement.

Supporting these additional values will greatly improve foot routing in well-mapped areas.

**Comment by matkoniecz** [comment]
> Assign the same low priority to ways tagged with sidewalk=separate. This tag means there is a sidewalk available, but it is mapped as a separate way. Thus the current way should not be used for routing.

It should be avoided, but sometimes due to incorrect mapping it is necessary to use it for routing anyway.

**Comment by cewbdex** [comment]
I think that osmand should also respect pedestrian crossing tags with sidewalk tags.

Also a small update: Streetcomplete app has now support for adding sidewalks, so I think those will get more used and should have higher priority for implementing. As having a separate way is not a viable option everywhere.

**Comment by scaidermern** [comment]
Yes. Foot routing should prefer to cross streets at nearby crossings. Crossing streets at other places could get a penalty, the penalty value could depend on the highway importance. So that crossing large streets without a crossing get's a high penalty.
